### ISSUE TITLE
CWL secondaryFiles need to be copied into working directories

### ISSUE THREAD

**Issue by kellrott** [BD2KGenomics/toil#615]
Secondary files defined in CWL files (as part of the inputBinding) need to be copied into place.
For example in the case of a tool that defines a BAM file, and its index .bai file:

```
  - id: "#input"
    type: File
    inputBinding:
      position: 1
      secondaryFiles:
        - .bai
```

Will give the error:

```
WARNING:toil.leader:Reporting file: Q/9/jobfpSkdv
WARNING:toil.leader:Q/9/jobfpSkdv:	---TOIL WORKER OUTPUT LOG---
WARNING:toil.leader:Q/9/jobfpSkdv:	INFO:rdflib:RDFLib Version: 4.2.1
WARNING:toil.leader:Q/9/jobfpSkdv:	Traceback (most recent call last):
WARNING:toil.leader:Q/9/jobfpSkdv:	  File "/Users/ellrott/workspaces/mc3_cwl/venv/lib/python2.7/site-packages/toil-3.2.0a2-py2.7.egg/toil/worker.py", line 282, in main
WARNING:toil.leader:Q/9/jobfpSkdv:	    fileStore=fileStore)
WARNING:toil.leader:Q/9/jobfpSkdv:	  File "/Users/ellrott/workspaces/mc3_cwl/venv/lib/python2.7/site-packages/toil-3.2.0a2-py2.7.egg/toil/job.py", line 1281, in _execute
WARNING:toil.leader:Q/9/jobfpSkdv:	    returnValues = self.run(fileStore)
WARNING:toil.leader:Q/9/jobfpSkdv:	  File "/Users/ellrott/workspaces/mc3_cwl/venv/lib/python2.7/site-packages/toil-3.2.0a2-py2.7.egg/toil/cwl/cwltoil.py", line 163, in run
WARNING:toil.leader:Q/9/jobfpSkdv:	    tmpdir=tmpdir)
WARNING:toil.leader:Q/9/jobfpSkdv:	  File "/Users/ellrott/workspaces/mc3_cwl/venv/lib/python2.7/site-packages/cwltool/main.py", line 159, in single_job_executor
WARNING:toil.leader:Q/9/jobfpSkdv:	    r.run(**kwargs)
WARNING:toil.leader:Q/9/jobfpSkdv:	  File "/Users/ellrott/workspaces/mc3_cwl/venv/lib/python2.7/site-packages/cwltool/job.py", line 51, in run
WARNING:toil.leader:Q/9/jobfpSkdv:	    raise WorkflowException("Required input file %s not found" % self.pathmapper.mapper(f)[0])
WARNING:toil.leader:Q/9/jobfpSkdv:	WorkflowException: Required input file /Users/ellrott/workspaces/mc3_cwl/work/tmpPZLXOr/localTempDir/tvwSkaZ/inp/testexome.pair0.normal.bam.bai not found
WARNING:toil.leader:Q/9/jobfpSkdv:	Exiting the worker because of a failed jobWrapper on host RJHB298
WARNING:toil.leader:Q/9/jobfpSkdv:	ERROR:toil.worker:Exiting the worker because of a failed jobWrapper on host RJHB298
WARNING:toil.leader:Q/9/jobfpSkdv:	WARNING:toil.jobWrapper:Due to failure we are reducing the remaining retry count of job Q/9/jobfpSkdv to 0
```

**Comment by hannes-ucsc** [comment]
@tetron, could you triage this? I'd just like to know what your take on this is?

**Comment by tetron** [comment]
I think this is the same issue I fixed in cwltool, I sent an email that the version pin needed to be updated (there was an email thread but I didn't have the issue #).  I will make a pull request with the latest version of cwltool, this is needed anyway to pull in draft-3 support.

**Comment by hannes-ucsc** [comment]
Resolved by #711
